src/components/builder/prefill/family-members/index.tsx

```diff
@@ -0,0 +1,63 @@
+import {JSONObject} from '@open-formulieren/types/lib/types';
+import {useContext} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import Select from '@/components/formio/select';
+import {BuilderContext} from '@/context';
+
+/**
+ * Get the available form variables and display them in a multiselect.
+ *
+ * This requires a function `getAvailableFormVariables` to be provided to the
+ * BuilderContext which is responsible for retrieving the list of available form variables.
+ */
+const UserDefinedFormVariableSelect: React.FC = () => {
+  const intl = useIntl();
+  const {getAvailableFormVariables} = useContext(BuilderContext);
+
+  const tooltip = intl.formatMessage({
+    description: "Tooltip for 'populateFrom' builder field",
+    defaultMessage:
+      'Select the user defined form variable from which this component will be prefilled.',
+  });
+
+  const availableFormVariables: JSONObject[] = getAvailableFormVariables();
+
+  // We only need to show the relevant variables that can be used for the Partners component
+  const _options: JSONObject[] = availableFormVariables
```

Comment:
In the task there is a bonus subtask which would filter the variables in the backend. I chose to do it here instead because I thought that the callback should be generic (just retrieve all the available variables) and do the special handling in the specific case where it's used. I would like your opinion.